contrib/google.golang.org/internal/grpcutil/tag_utils_test.go

```diff
@@ -0,0 +1,63 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package grpcutil
+
+import (
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+)
+
+func TestExtractTags(t *testing.T) {
+	testCases := []struct {
+		Name    string
+		Path    string
+		Service string
+		Method  string
+		Package string
+	}{
+		{"empty test",
+			"",
+			"",
+			"",
+			"",
+		},
+		{"basic test",
+			"/mypackage.myservice/mymethod",
+			"mypackage.myservice",
+			"mymethod",
+			"mypackage",
+		},
+		{"obscure test",
+			"/my/p/a/c.k.a.ge.my/se/r/v/ice/myme.t.h.od",
```

Comment:
for non-valid grpc full_method names, I think the expected result should be "unknown" since these tags are mandatory but we just somehow got an unexpected string.

Reply:
I think these are valid grpc names, since I followed the rules from [Otel](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/rpc.md#span-name) where the service does not have periods and the method does not have slashes. Everything else is fair game as far as am I aware.